Assets/Scripts/Game/GameManager.cs

```diff
@@ -11,22 +11,24 @@ namespace Game
 {
     public class GameManager : MonoBehaviour
     {
-        public GameObject _spawner;
-        public int oilBottleCountForSpawn;
+        [SerializeField] private float _wallWarningInterval;
+        [SerializeField] private float _wallDangerousInterval;
+        [SerializeField] private float _wallCoolDownInterval;
+        [SerializeField] private GameObject _spawner;
+        [SerializeField] private int oilBottleCountForSpawn;
 
+        private List<WallDanger> _restingWalls = new List<WallDanger>();
+        private List<WallDanger> _wallsToBeDangerous;
         private GameWaveManager _waveManager;
         private Score _score;
-        private static List<GameObject> WallsToBeDangerous => FindObjectsOfType<WallDanger>().Select(w => w.gameObject).ToList();
+        private int _dangerWallsCurrently;
```

Comment:
This should be a local variable.

Reply:
It is not static at the last commit and it is used in different coroutines now

Reply:
Agree